OneCommunityGlobal/HighestGoodNetworkApp
# Tim fix dashboard badge modal

# Description
(PRIORITY MEDIUM) Fix View Full Badge History modal (Dashboard > Badges > Badge Report button) modal to not crash app. Attempting to edit values in the modal and clicking Save Changes results in an error that freezes the app and forces a page refresh.

<img width="396" alt="app-crash-error" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/f20e6a6d-2a5f-4e54-bc5b-c7b1ad7f8fb5">

## Related PRS (if any):
N/A

## Main changes explained:
- The Badge Summary modal accessible on the Dashboard (Dashboard > Badges > Badge Report) has been changed from editable to the same static modal that displays on the People Reports page.
   - The crashes were caused by unavailable (because the necessary prop drilling had not been set up) functions being called.
- All Badge Summary modals (Dashboard, People Reports, User Profile) have been updated to show text if the user has no badges.
   - “You have no badges” if it is the user’s own report.
   - “This person has no badges” if it is another user’s report.
- `key` prop errors in the badge summary modals have been fixed.
- A `button` element has been replaced by a `div` to fix a nested button error.

<img width="392" alt="nested-button-error" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/d1640f73-6cd3-4199-a04e-1f7482b3a983">

## How to test:
- `git checkout Tim_fix_dashboard_badge_modal` to check into current branch
- `npm install` to ensure your modules are up to date (no new modules are added in this PR)
- `npm run start:local` to run this PR locally
- Log in as an admin
- Navigate to Dashboard > Badges > Badge Report. The modal should not be editable and should not crash the app when closed. There should be no “key prop” or “nested button” errors in the console.
- In the User Profile > Select Featured modal you can temporarily delete your badges to check the text “You have no badges.” is displayed. You can test against various screen widths as well. Also check that this modal does not display “key prop” or “nested button” errors in your console.
- Lastly, use Other Links > User Management to navigate to a different user account who has no badges (you can delete the badges from one of your own test accounts, or use Tim Core). The text displayed should now be “This person has no badges.”
- You can run the same tests on the badge modal located at Reports > Reports > People > select a person > Show Badges.  

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/4cd85749-56f1-43ab-916c-9df33027ff44
